File: ios/MyMathLib2.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MyMathLib2, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a 
                  withB:(float)b
                  withResolver:(RCTPromiseResolveBlock)resolve 
                  withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(add:(float)a 
                 withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve 
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(subtract:(float)a 
                    withB:(float)b 
                    withResolver:(RCTPromiseResolveBlock)resolve 
                    withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(divide:(float)a 
                  withB:(float)b 
                  withResolver:(RCTPromiseResolveBlock)resolve 
                  withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/MyMathLib2.swift
@objc(MyMathLib2)
class MyMathLib2: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        resolve(a * b)
    }

    @objc(add:withB:withResolver:withRejecter:)
    func add(a: Float, b: Float, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        resolve(a + b)
    }

    @objc(subtract:withB:withResolver:withRejecter:)
    func subtract(a: Float, b: Float, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        resolve(a - b)
    }

    @objc(divide:withB:withResolver:withRejecter:)
    func divide(a: Float, b: Float, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        if b == 0 {
            reject("DIVIDE_BY_ZERO", "Cannot divide by zero", nil)
        } else {
            resolve(a / b)
        }
    }
}
